### ISSUE TITLE
Support HEIC Image Format (auto-convert)

### ISSUE THREAD

**Issue by abegehr** [FlutterFlow/flutterflow-issues#4642]
### Can we access your project?

- [X] I give permission for members of the FlutterFlow team to access and test my project for the sole purpose of investigating this issue.

### Current Behavior

When uploading a HEIC (heigh quality) image anywhere in the FF app (f.e. on the user profile page), it fails with error: "Invalid file format: image/heic".
![Screenshot 2024-10-25 at 10 50 38](https://github.com/user-attachments/assets/0331eb72-ab34-441b-9c47-25a8ab3dbeb8)

### Expected Behavior

If the user uploads a HEIC image, it should be supported or auto-converted into a supported format.
If that's not feasible, the "Upload media to Firebase" action should allow filtering out HEIC images in the photo picker.

### Steps to Reproduce

1. Connect Firebase project.
2. Create new page
3. Add Button
4. On Tap, add a "Upload Image to Firebase" action
5. Run the app on an Android with HEIC images and try to select a HEIC-format image.
6. See the error snacker from screenshot.

### Reproducible from Blank

- [X] The steps to reproduce above start from a blank project.

### Bug Report Code (Required)

ITFXlvHqx892rsBE15PIbcdviWIwFj8bbZI7iO4YGEgbIYj1OpY2ZvT+aXdaXseadws7B0yL+VgEp/TliuOeU8ICFEisRb1607oJSzvJTnqmbJitC8+CXnxPN+9PIBWO0MKNryUlJsdvSGdkw0+Lcq3qNleeY8aSfxBlZ7vfcPo=

### Visual documentation

![Screenshot 2024-10-25 at 10 50 38](https://github.com/user-attachments/assets/1b960e7c-a786-4acb-bf48-493d8f276070)

### Environment

```markdown
- FlutterFlow version: v5.0.8+ released October 22, 2024
- Platform: Android Samsung Galaxy S10 lite SM-G770F
- Browser name and version: Chrome 130.0.6723.70 (Official Build) (arm64)
- Operating system and version affected: Android Version 13
- Flutter version: 3.24.2
```

**Comment by ghost** [comment]
yes please! this would be sweet

**Comment by Alezanello** [comment]
Hello!

You’re correct; currently, this action supports four formats (jpg, png, gif, and mp4). I can check with the team to see if there are any plans to support additional formats. Since this behavior is expected, I’ll close this issue as a feature request and create an internal ticket for it. Rest assured, the engineering team will be made aware of your feedback!

**Comment by ShaneVC01** [comment]
How can this not be planned? More than 90% of my users access my application on an iOS device. I already created the function to auto convert from HEIC to JPG. The issue is that the supported file types does not include HEIC.

The way around this for iPhone is to change using the "ANY" for "Upload/Save File". This however has another impact. I expect users to upload images or videos to my application. Changing from "Upload/Save Media" to "Upload/Save File", force the iPhone to browse for Files, and not photos or videos from their Gallery.

Can we at least get an option to include HEIC as part of the supported file types and handle the conversion ourselves? As an example, I allow user to select HEIC, then convert to JPG and store the JPG file in Firebase Storage.

**Comment by abegehr** [comment]
Yes, definitely not great, as FF targets iOS and Android, so users would expect to be able to upload all sorts of pictures from photo library.

**Comment by Alezanello** [comment]
Thank you for bringing this up! I apologize if my previous response caused any confusion. The issue was marked as "Not Planned" because this GitHub Issue Tracker is intended specifically for bugs, and your request falls under a feature or enhancement request rather than a bug.

Rest assured, I’ve created an internal ticket with your request, and it's now in our Engineering Team's backlog. We’ve also included a link back to this issue so the team can review your detailed request and understand the need for HEIC support.

In summary, while this issue has been closed as "Not Planned" here on GitHub to keep this tracker focused on bugs, your request for HEIC support is documented internally and will be considered by the team. Thank you for your understanding!
